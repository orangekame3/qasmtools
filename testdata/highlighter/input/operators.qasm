OPENQASM 3.0;

// Arithmetic operators
let a = 1 + 2;
let b = 3 - 4;
let c = 5 * 6;
let d = 7 / 8;

// Comparison operators
if (a == b) {
    let x = 1;
}

// Quantum operators
cx q[0], q[1];
h q[0];
measure q -> c;
