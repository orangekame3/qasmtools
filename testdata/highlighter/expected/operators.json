{
	"tokens": [
		{
			"type": "keyword",
			"content": "OPENQASM",
			"line": 1,
			"column": 0
		},
		{
			"type": "number",
			"content": "3.0",
			"line": 1,
			"column": 9
		},
		{
			"type": "punctuation",
			"content": ";",
			"line": 1,
			"column": 12
		},
		{
			"type": "comment",
			"content": "// Arithmetic operators",
			"line": 3,
			"column": 0
		},
		{
			"type": "keyword",
			"content": "let",
			"line": 4,
			"column": 0
		},
		{
			"type": "identifier",
			"content": "a",
			"line": 4,
			"column": 4
		},
		{
			"type": "operator",
			"content": "=",
			"line": 4,
			"column": 6
		},
		{
			"type": "number",
			"content": "1",
			"line": 4,
			"column": 8
		},
		{
			"type": "operator",
			"content": "+",
			"line": 4,
			"column": 10
		},
		{
			"type": "number",
			"content": "2",
			"line": 4,
			"column": 12
		},
		{
			"type": "punctuation",
			"content": ";",
			"line": 4,
			"column": 13
		},
		{
			"type": "keyword",
			"content": "let",
			"line": 5,
			"column": 0
		},
		{
			"type": "identifier",
			"content": "b",
			"line": 5,
			"column": 4
		},
		{
			"type": "operator",
			"content": "=",
			"line": 5,
			"column": 6
		},
		{
			"type": "number",
			"content": "3",
			"line": 5,
			"column": 8
		},
		{
			"type": "operator",
			"content": "-",
			"line": 5,
			"column": 10
		},
		{
			"type": "number",
			"content": "4",
			"line": 5,
			"column": 12
		},
		{
			"type": "punctuation",
			"content": ";",
			"line": 5,
			"column": 13
		},
		{
			"type": "keyword",
			"content": "let",
			"line": 6,
			"column": 0
		},
		{
			"type": "identifier",
			"content": "c",
			"line": 6,
			"column": 4
		},
		{
			"type": "operator",
			"content": "=",
			"line": 6,
			"column": 6
		},
		{
			"type": "number",
			"content": "5",
			"line": 6,
			"column": 8
		},
		{
			"type": "operator",
			"content": "*",
			"line": 6,
			"column": 10
		},
		{
			"type": "number",
			"content": "6",
			"line": 6,
			"column": 12
		},
		{
			"type": "punctuation",
			"content": ";",
			"line": 6,
			"column": 13
		},
		{
			"type": "keyword",
			"content": "let",
			"line": 7,
			"column": 0
		},
		{
			"type": "identifier",
			"content": "d",
			"line": 7,
			"column": 4
		},
		{
			"type": "operator",
			"content": "=",
			"line": 7,
			"column": 6
		},
		{
			"type": "number",
			"content": "7",
			"line": 7,
			"column": 8
		},
		{
			"type": "operator",
			"content": "/",
			"line": 7,
			"column": 10
		},
		{
			"type": "number",
			"content": "8",
			"line": 7,
			"column": 12
		},
		{
			"type": "punctuation",
			"content": ";",
			"line": 7,
			"column": 13
		},
		{
			"type": "comment",
			"content": "// Comparison operators",
			"line": 9,
			"column": 0
		},
		{
			"type": "keyword",
			"content": "if",
			"line": 10,
			"column": 0
		},
		{
			"type": "punctuation",
			"content": "(",
			"line": 10,
			"column": 3
		},
		{
			"type": "identifier",
			"content": "a",
			"line": 10,
			"column": 4
		},
		{
			"type": "operator",
			"content": "==",
			"line": 10,
			"column": 6
		},
		{
			"type": "identifier",
			"content": "b",
			"line": 10,
			"column": 9
		},
		{
			"type": "punctuation",
			"content": ")",
			"line": 10,
			"column": 10
		},
		{
			"type": "punctuation",
			"content": "{",
			"line": 10,
			"column": 12
		},
		{
			"type": "keyword",
			"content": "let",
			"line": 11,
			"column": 4
		},
		{
			"type": "identifier",
			"content": "x",
			"line": 11,
			"column": 8
		},
		{
			"type": "operator",
			"content": "=",
			"line": 11,
			"column": 10
		},
		{
			"type": "number",
			"content": "1",
			"line": 11,
			"column": 12
		},
		{
			"type": "punctuation",
			"content": ";",
			"line": 11,
			"column": 13
		},
		{
			"type": "punctuation",
			"content": "}",
			"line": 12,
			"column": 0
		},
		{
			"type": "comment",
			"content": "// Quantum operators",
			"line": 14,
			"column": 0
		},
		{
			"type": "identifier",
			"content": "cx",
			"line": 15,
			"column": 0
		},
		{
			"type": "identifier",
			"content": "q",
			"line": 15,
			"column": 3
		},
		{
			"type": "punctuation",
			"content": "[",
			"line": 15,
			"column": 4
		},
		{
			"type": "number",
			"content": "0",
			"line": 15,
			"column": 5
		},
		{
			"type": "punctuation",
			"content": "]",
			"line": 15,
			"column": 6
		},
		{
			"type": "punctuation",
			"content": ",",
			"line": 15,
			"column": 7
		},
		{
			"type": "identifier",
			"content": "q",
			"line": 15,
			"column": 9
		},
		{
			"type": "punctuation",
			"content": "[",
			"line": 15,
			"column": 10
		},
		{
			"type": "number",
			"content": "1",
			"line": 15,
			"column": 11
		},
		{
			"type": "punctuation",
			"content": "]",
			"line": 15,
			"column": 12
		},
		{
			"type": "punctuation",
			"content": ";",
			"line": 15,
			"column": 13
		},
		{
			"type": "identifier",
			"content": "h",
			"line": 16,
			"column": 0
		},
		{
			"type": "identifier",
			"content": "q",
			"line": 16,
			"column": 2
		},
		{
			"type": "punctuation",
			"content": "[",
			"line": 16,
			"column": 3
		},
		{
			"type": "number",
			"content": "0",
			"line": 16,
			"column": 4
		},
		{
			"type": "punctuation",
			"content": "]",
			"line": 16,
			"column": 5
		},
		{
			"type": "punctuation",
			"content": ";",
			"line": 16,
			"column": 6
		},
		{
			"type": "measurement",
			"content": "measure",
			"line": 17,
			"column": 0
		},
		{
			"type": "identifier",
			"content": "q",
			"line": 17,
			"column": 8
		},
		{
			"type": "operator",
			"content": "->",
			"line": 17,
			"column": 10
		},
		{
			"type": "identifier",
			"content": "c",
			"line": 17,
			"column": 13
		},
		{
			"type": "punctuation",
			"content": ";",
			"line": 17,
			"column": 14
		}
	]
}
